--- /tmp/SOAP-Lite-0.715/blib/lib/SOAP/Transport/HTTP.pm	2012-07-15 11:18:44.000000000 +0200
+++ lib/SOAP/Transport/HTTP.pm	2012-09-03 11:26:13.742952521 +0200
@@ -317,9 +317,14 @@
 "Can't understand returned Content-Encoding (@{[$self->http_response->content_encoding]})\n"
       : $self->http_response->content;
 
-    return $self->http_response->content_type =~ m!^multipart/!i
+    $content = $self->http_response->content_type =~ m!^multipart/!i
       ? join( "\n", $self->http_response->headers_as_string, $content )
       : $content;
+
+    # untaint
+    $content =~ /^(.*)$/s;
+    $content = $1;
+    return $content;
 }
 
 # ======================================================================
